### ISSUE TITLE
v1 Deprecation, pull `wait` out of actions

### ISSUE THREAD

**Issue by AustinAbro321** [zarf-dev/zarf#2640]
`actions.wait` will be deprecated in favor of a wait list per component that is run on deploy. `action.wait` blocks that are `onDeploy.After` can be auto migrated and will work if deployed on v1. `wait` blocks that are not on `onDeploy.After` will not be auto migrated and the package will error out if deployed on v1. Zarf v1 will use kstatus for wait functionality instead of `kubectl wait`. `apiVersion` will be added as a required field on `wait`, and the `apiVersion` will be automatically set to the most recent `apiVersion` if the waited for resource is not a custom resource. Users can still wait on custom resources, but the resources must implement kStatus. `wait.condition` will be removed from the schema and Zarf will assume it is waiting for `ready`. A `wait` block will not be auto migrated if `condition` is not `set` to  `ready`, `available` or equivalent.
In summary, a `wait` block will be auto migrated if, and only if:
  - it is `onDeploy.after`
  - the condition is `ready`, `available` or equivalent
  - The kind is not a custom resource.

**Comment by AustinAbro321** [comment]
Closing this issue for now. It has not been decided how exactly we will handle actions.wait.cluster in the new schema, but it will not be as described above
